File: hybrid/lib/entities/pagination.dart
import 'package:json_annotation/json_annotation.dart';

part 'pagination.g.dart';

@JsonSerializable()
class Pagination {
  int currentPage;
  int totalPages;
  int pageSize;
  int totalCount;

  Pagination(
      {required this.currentPage,
      required this.totalPages,
      required this.pageSize,
      required this.totalCount});

  factory Pagination.fromJson(Map<String, dynamic> json) =>
      _$PaginationFromJson(json);

  Map<String, dynamic> toJson() => _$PaginationToJson(this);
}

class PaginatedResponse {
  dynamic items;
  Pagination pagination;

  PaginatedResponse({required this.items, required this.pagination});
}

class PaginatedFilter {
  int pageNumber;
  int pageSize;
  PaginatedFilter({required this.pageNumber, required this.pageSize});
}

File: client/src/app/shared/models/pagination.ts
export interface MetaData {
  currentPage: number;
  totalPages: number;
  pageSize: number;
  totalCount: number;
}

export class PaginatedResponse<dynamic> {
  items: dynamic;
  metaData: MetaData;

  // constructor(items: T, metaData: MetaData) {
  //     this.items = items;
  //     this.metaData = metaData;
  // }
}

export interface PaginatedFilter {
  pageNumber: number;
  pageSize: number;
}
